src/test/java/bio/terra/service/snapshot/SnapshotDaoTest.java

```diff
@@ -350,12 +351,19 @@ private void testOneEnumerateRange(List<UUID> snapshotIds,
         List<SnapshotSummary> summaryList = summaryEnum.getItems();
         int index = offset;
         for (SnapshotSummary summary : summaryList) {
+            Snapshot fromDB = snapshotDao.retrieveSnapshot(snapshotIds.get(index));
+            SnapshotSource source = fromDB.getFirstSnapshotSource();
             assertThat("correct snapshot id",
                 snapshotIds.get(index),
                 equalTo(summary.getId()));
             assertThat("correct snapshot name",
                 makeName(snapshotName, index),
                 equalTo(summary.getName()));
+            assertThat("snapshot includes dataset storage regions",
+                    source.getDataset().getDatasetSummary().getStorage().stream()
+                            .allMatch(sr -> sr.getRegion().equals(GoogleRegion.US_CENTRAL1)),
+                    equalTo(summary.getStorage().stream()
+                            .allMatch(sr -> sr.getRegion().equals(GoogleRegion.US_CENTRAL1))));
```

Comment:
I think this test could be better written using some other Hamcrest matchers. As is stands now, we're only checking that both the dataset summary and snapshot summary have all their storage resources in `US_CENTRAL1`. In this check, a snapshot could omit one of the storage resources, but the test would still pass because the rest of the storage resources would still be in `US_CENTRAL1`

Reply:
That's a good point!